test/datasets/speechcommands_test.py

```diff
@@ -0,0 +1,70 @@
+import os
+
+from torchaudio.datasets import speechcommands
+
+from ..common_utils import (
+    TempDirMixin,
+    TorchaudioTestCase,
+    get_whitenoise,
+    normalize_wav,
+    save_wav,
+)
+
+
+class TestSpeechCommands(TempDirMixin, TorchaudioTestCase):
+    backend = "default"
+
+    root_dir = None
+    samples = []
+
+    @classmethod
+    def setUp(cls):
+        cls.root_dir = cls.get_base_temp_dir()
+        dataset_dir = os.path.join(
+            cls.root_dir, speechcommands.FOLDER_IN_ARCHIVE, speechcommands.URL
+        )
+        os.makedirs(dataset_dir, exist_ok=True)
+        sample_rate = 16000  # 16kHz sample rate
+        for i in range(30):
+            label = "label_{:02d}".format(i)
```

Comment:
Directory structure must mimic that of the real dataset. That is, you need to make directory with names like the following; 

<details>

```
bed
bird
cat
dog
down
eight
five
follow
forward
four
go
happy
house
learn
left
marvin
nine
no
off
on
one
right
seven
sheila
six
stop
three
tree
two
up
visual
wow
yes
zero
```

</details>

these are the supposed label values returned by dataset.

also note that Dataset will walk directories in alphabetical order so you need to add samples in the expected order too.